server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java

```diff
@@ -0,0 +1,531 @@
+import org.apache.accumulo.core.dataImpl.thrift.TColumn;
+import org.apache.accumulo.core.dataImpl.thrift.TKeyExtent;
+import org.apache.accumulo.core.dataImpl.thrift.TKeyValue;
+import org.apache.accumulo.core.dataImpl.thrift.TRange;
+import org.apache.accumulo.core.sample.impl.SamplerConfigurationImpl;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.securityImpl.thrift.TCredentials;
+import org.apache.accumulo.core.spi.scan.ScanDispatcher;
+import org.apache.accumulo.core.tabletserver.thrift.ActiveScan;
+import org.apache.accumulo.core.tabletserver.thrift.NoSuchScanIDException;
+import org.apache.accumulo.core.tabletserver.thrift.NotServingTabletException;
+import org.apache.accumulo.core.tabletserver.thrift.TSampleNotPresentException;
+import org.apache.accumulo.core.tabletserver.thrift.TSamplerConfiguration;
+import org.apache.accumulo.core.tabletserver.thrift.TabletScanClientService;
+import org.apache.accumulo.core.trace.thrift.TInfo;
+import org.apache.accumulo.core.util.Halt;
+import org.apache.accumulo.fate.zookeeper.ServiceLock;
+import org.apache.accumulo.fate.zookeeper.ZooUtil;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.fs.TooManyFilesException;
+import org.apache.accumulo.server.rpc.TServerUtils;
+import org.apache.accumulo.server.security.AuditedSecurityOperation;
+import org.apache.accumulo.server.security.SecurityOperation;
+import org.apache.accumulo.tserver.scan.LookupTask;
+import org.apache.accumulo.tserver.scan.NextBatchTask;
+import org.apache.accumulo.tserver.scan.ScanParameters;
+import org.apache.accumulo.tserver.session.MultiScanSession;
+import org.apache.accumulo.tserver.session.SingleScanSession;
+import org.apache.accumulo.tserver.tablet.KVEntry;
+import org.apache.accumulo.tserver.tablet.ScanBatch;
+import org.apache.accumulo.tserver.tablet.Tablet;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Collections2;
+
+public class ThriftScanClientHandler implements TabletScanClientService.Iface {
+
+  private static final Logger log = LoggerFactory.getLogger(ThriftScanClientHandler.class);
+
+  private final TabletServer server;
+  protected final ServerContext context;
+  protected final SecurityOperation security;
+  private final WriteTracker writeTracker;
+  private final long MAX_TIME_TO_WAIT_FOR_SCAN_RESULT_MILLIS;
+
+  public ThriftScanClientHandler(TabletServer server, WriteTracker writeTracker) {
+    this.server = server;
+    this.context = server.getContext();
+    this.writeTracker = writeTracker;
+    this.security = AuditedSecurityOperation.getInstance(context);
+    MAX_TIME_TO_WAIT_FOR_SCAN_RESULT_MILLIS = server.getContext().getConfiguration()
+        .getTimeInMillis(Property.TSERV_SCAN_RESULTS_MAX_TIMEOUT);
+  }
+
+  private void checkPermission(TCredentials credentials, String lock, final String request)
```

Comment:
Is this method identical to what is in TabletClientHandler?  If so may be good to share the code.  Maybe once class could have a static method that they both call.

Reply:
Yes, should be a copy. I can consolidate.

Reply:
Addressed in d4b1a76